internal/mcpproxy/handlers.go

```diff
@@ -664,6 +757,32 @@ func (m *MCPProxy) proxyResponseBody(ctx context.Context, s *session, w http.Res
 							continue
 						}
 						msg.ID = req.ID
+
+						// Check if this is a JSON-RPC error response
+						if msg.Error != nil {
+							// Collect error to return to caller
+							responseErrors = append(responseErrors, msg.Error)
+						}
+
+						// Check if this is a tools/call response with isError=true
+						if req.Method == "tools/call" && msg.Result != nil {
```

Comment:
This code is very similar to the code above. refactor it to a small method to favor code reuse.